test/services/service_status.service.test.js

```diff
@@ -3,38 +3,176 @@
-  describe('importRows data', () => {
-    it('returns data in the expected format', async () => {
-      const { importRows } = await ServiceStatusService.go()
+  beforeEach(() => {
+    // These requests will remain unchanged throughout the tests. We do alter the ones to the AddressFace and the
+    // water-api (foreground-service) though, which is why they are defined separately in each test.
+    Nock('http://localhost:8020')
+      .get('/status')
+      .reply(200, { status: 'alive' },
+        [
+          'x-cma-git-commit', '273604040a47e0977b0579a0fef0f09726d95e39',
+          'x-cma-docker-tag', 'ghcr.io/defra/sroc-charging-module-api:v9.99.9'
+        ]
+      )
 
-      expect(importRows).to.equal([
-        firstRow,
-        secondRow
+    Nock('http://localhost:8012').get('/health/info').reply(200, { version: '8.0.12', commit: '83d0e8c' })
+    Nock('http://localhost:8011').get('/health/info').reply(200, { version: '8.0.11', commit: 'a7030dc' })
+    Nock('http://localhost:8007').get('/health/info').reply(200, { version: '8.0.7', commit: 'a181fb1' })
+    Nock('http://localhost:8002').get('/health/info').reply(200, { version: '8.0.2', commit: '58bd0c1' })
+    Nock('http://localhost:8000').get('/health/info').reply(200, { version: '8.0.0', commit: 'f154e3f' })
+    Nock('http://localhost:8008').get('/health/info').reply(200, { version: '8.0.8', commit: 'f154e3f' })
+    Nock('http://localhost:8003').get('/health/info').reply(200, { version: '8.0.3', commit: '2ddff3c' })
+    Nock('http://localhost:8004').get('/health/info').reply(200, { version: '8.0.4', commit: '09d5261' })
+    Nock('http://localhost:8006').get('/health/info').reply(200, { version: '8.0.6', commit: 'b181625' })
+  })
+
+  afterEach(() => {
+    Sinon.restore()
+    Nock.cleanAll()
+  })
+
+  describe('when all the services are running', () => {
+    beforeEach(async () => {
+      // In this scenario everything is hunky-dory so we return 2xx responses from these services
+      Nock('http://localhost:8009').get('/address-service/hola').reply(200, 'hey there')
+      Nock('http://localhost:8001').get('/health/info').reply(200, { version: '8.0.1', commit: '83d0e8c' })
+
+      // Unfortunately, this convoluted test setup is the only way we've managed to stub how the promisified version of
+      // `child-process.exec()` behaves in the class under test.
+      // We create an anonymous stub, which responds differently for the 1st and 2nd call. We rely on knowing the order
+      // they are called in the class under test for this to work.
+      // We then stub the util library's `promisify()` method and tell it to calll our anonymous stub when invoked.
+      // The bit that makes all this work is the fact we use Proxyquire to load our stubbed util instead of the real
+      // one when we load our class under test
+      const execStub = Sinon.stub().onFirstCall().resolves({ stdout: 'ClamAV 9.99.9/26685/Mon Oct 10 08:00:01 2022\n', stderror: null })
```

Comment:
I haven't looked into this too deeply and maybe it's already been considered, but could it be possible to use `stub.withArgs()` in conjunction with a matcher like `sinon.match(string)` to stub specific calls?

Reply:
Not sure how `sinon.matcher()` applies. But `withArgs()` was a winner! Have submitted a change. Let me know what you think.